packages/auth/src/Auth.ts

```
@@ -1187,15 +1368,24 @@ export default class AuthClass {
             logger.debug('failed to clear cached items');
         }
 
-        if (this.userPool) { 
+        await this._storageSync;
+        const sessionSource = this._getSessionSource();
+        this._removeSessionSource();
+        if (!sessionSource || sessionSource === AWSCognitoProvider.NAME) {
+            if (!this.userPool) {
+                // throw('No userPool'); 

```

Is this intentional? should it be removed/uncommented?

---

Good catch. We should throw the error when user pool is not configured.